### DISCUSSION TITLE
Changing Start Point from Favorite to Map Position

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#181] [Q&A] [answered]
When you plan a route in Cruiser app using a Favorite as a Start Point, if you then make a new Start Point using the map - the Favorite "name" is carried-over as the new starting point.  Although it doesn't affect routing, I'm not sure it's *ideal behaviour* :)

It happens on Desktop, too - Here is the result...
1st: Plot route from Favorite to a short distance away
2nd: Right-click map and set new starting point a bit further back
3rd: Right-click Favorite as a Via-Point
![start-point-name](https://github.com/devemux86/cruiser/assets/125696109/1d81b94c-3da6-4687-a35a-bfd7ca4aa498)

This also raises the idea of map "knowing" what road name I'm selecting when creating a Shaping-Point.  I've touched on this elsewhere regarding how BaseCamp does it - it doesn't allow duplicate names of *anything*, so the M25 motorway could have shaping-points called M251, M252, M253 etc across your collection.  Same with a name that you give a Favourite/Waypoint, like: "Fuel, Shell".  You could only have 1 of those unless you include a location or let it suffix them numerically.  Fun!

So, if I had a route that required a Via-Point, without resorting to the "Edit" menu, is there a way for Cruiser to obtain the road name (or nearest geographical location)?

**Comment by devemux86** [comment]
@MotoUKRider thanks for the report.

Creating waypoints from existing map elements (favorites, search results, etc.)
uses the name of these map elements as new name on the created waypoints.

Also the "Set start" for an existing start waypoint works like when we move it.
It only changes its coordinates, it does not create a new waypoint / name.

This was discussed in the past and was preferred by the users
in order to avoid losing their custom names on the waypoints when move them.

> how BaseCamp does it - it doesn't allow duplicate names of anything

Why is this good? Doesn't it remove flexibility from users?
As it does not allow them to do whatever they want with their software?

Or maybe you mean something else?

> if I had a route that required a Via-Point, without resorting to the "Edit" menu, is there a way for Cruiser to obtain the road name (or nearest geographical location)?

This requires reverse geocode of the coordinates and it is only an _online_ process.
Also it returns the nearest _snapped_ geocoded location, not the clicked map point.

You can test it, if you right click (or long press) on map and select "Show address".

**Reply by MotoUKRider** [reply]
> Also the "Set start" for an existing start waypoint works like when we move it. It only changes its coordinates, it does not create a new waypoint / name. This was discussed in the past and was preferred by the users in order to avoid losing their custom names on the waypoints when move them.

Interesting - so, in my scenario, I'd have to use the "Edit" function or clear the route and start again?  My concern is that the newly placed start-point is *not* the Favorite name previously used.  Perhaps a popup dialog like: *"Do you want to keep xxxx as the Starting Point Name?"*

> Why is this good? Doesn't it remove flexibility from users? As it does not allow them to do whatever they want with their software?
> Or maybe you mean something else?

I offered it purely for context, but I've learnt to live with it and it does mean that I sometimes have get creative with a particular Shaping-Point as I can't have repeatable names like *"Turn off here"* in different places.  So yes, restrictive in some ways - and certainly makes the road-names a bit strange, for example here - the Shaping-Points all have "creative" names automatically applied:

![basecamp-route-list](https://github.com/devemux86/cruiser/assets/125696109/b066b9c4-49a2-4ce5-a4a9-ba5813e67e6b)

If you look closely, a couple of those are in the wrong numerical order - meaning that I plotted the *higher* ones later but they happen to fall *earlier* in the route.

**Comment by devemux86** [comment]
Move waypoint does not delete its name, as it would be strange to lose it.
We often move waypoints to adjust better their location, near or far away.

However the question in the past was what to do with "Set Start / End".
Keep the current name or delete it?

It is better to avoid extra dialogs if we can find an answer that most users expect and want?
What do you all think about it?

**Reply by MotoUKRider** [reply]
For clarity, I'm only referring to when you select a *Favorite* as a Starting-Point and then re-select a new Starting-Point using the cross-hairs on the map.

**Reply by devemux86** [reply]
Addresses may be similar: we create a waypoint with an address.
Then we move the waypoint away from the address location.
Should the waypoint name be deleted automatically?

There is no distinction for name source: favorite, address or custom name.
Also the offline formats (gpx, kurviger) only store the name, not its source.

So if we change the workflow, it will work globally for all name sources.

**Reply by DolomitesTom** [reply]
I vote for not losing names when moving WPs.

**Reply by devemux86** [reply]
> I vote for not losing names when moving WPs.

What if there is already a start / end and we "Set start / end" in another location?
Should the name of start / end be kept (like now) or reset?

And it is not only the names, but also the weights (beeline) that are currently kept.

**Reply by 0709wiwiwi** [reply]
It is a just custom to add a name start anyway at the Green start pin Via point.
But the green start pin should not necessarily be given a name. Suppose you have manually assigned a name such as "Tour start". 
If the green start pin position is then moved, you will than get a Via Point position with the informative name "Tourstart".

**Reply by DolomitesTom** [reply]
In this case name replacement is better

**Reply by 0709wiwiwi** [reply]
@DolomitesTom.
To repeat. 
By Cruiser standard in a gpx navtrack the green Via start pin is not automatically given a name.  
![Simonskall](https://github.com/devemux86/cruiser/assets/54167201/c3392874-8333-4425-9ef6-fd892f3aa72d)

So you do than plan a motorbike tour that you will ride together with some friends, right ?  
The name given by you for the green Via start pin or meeting point is "Start Eifel tour".
![Simonskall start Eifel tour](https://github.com/devemux86/cruiser/assets/54167201/d3216117-0259-4b6b-a5a4-04e92f603d40)

Cruiser calculates from each riders home position the missing bit navtrack towards "Start Eifel tour".
Just before arriving at the meeting point this Via point position is than announced as "Start Eifel tour".
A standard green Via start pin without any name tag would simply be announced as "Waypoint".

**Reply by devemux86** [reply]
It seems only the case where we use the "Set start / end" again (while start / end already exist) should be discussed?
To reset their names in case of a new "Set" action.

**Comment by devemux86** [comment, marked as answer]
In [Cruiser 3.0.22](https://play.google.com/store/apps/details?id=gr.talent.cruiser) the actions "Set start" and "Set end" place new waypoints.

**Reply by DolomitesTom** [reply]
this looks good and works in principle, but not in all cases: I had again the same effect as once in the past when there was no shaping or waypoint between the start and a not planned stay in a supermarket to eat and buy something. Even if I always have set "strict routing", it deletes over 100 sps and shows only the destination to drive or skip to. It even shows this "skip multiple" or "start at each sp" ONLY ONE TIME. It doesn't ask if I want to do this. And even if I then load again the complete route from file system, it doesn't ask again where I want to start. It just deletes all SPs and want me to drive to my destination, but I have not even started my tour. It took a while until I recalled this situation in the past. I guess this also can happen to others and then you ask yourself, why the shit is it doing this :-)  .... sure there is a workaround to set a sp behind your actual gps location, but you have to know it. Can't this be prevented somehow?

**Reply by devemux86** [reply]
@DolomitesTom this is more about the "Set start" / "Set end" actions in route planning.

I am not sure to understand clearly how is this related to navigation?

If there is an issue, can you please post a route example (with user location) and all detailed steps,
so we can reproduce the report and see what happens?

**Reply by DolomitesTom** [reply]
have sent en email for privacy reasons
